c7n/resources/athena.py

```diff
@@ -104,3 +106,46 @@ def process(self, resources):
         client = local_session(self.manager.session_factory).client("athena")
         for r in self.filter_resources(resources, "Status", ("ACTIVE", "PENDING")):
             client.cancel_capacity_reservation(Name=r["Name"])
+
+
+@resources.register("athena-query-execution")
+class AthenaQueryExecution(query.QueryResourceManager):
+    class resource_type(query.TypeInfo):
+        service = "athena"
+        enum_spec = ("list_query_executions", "QueryExecutionIds", None)
+        detail_spec = ("get_query_execution", "QueryExecutionId", None, "QueryExecution")
+        batch_detail_spec = (
+            "batch_get_query_execution",
+            "QueryExecutionIds",
+            None,
+            "QueryExecutions",
+            None,
+        )
+        arn = False
+        id = "QueryExecutionId"
+        name = "QueryExecutionId"
+
+
+@AthenaQueryExecution.filter_registry.register("runtime-statistics")
+class RuntimeStatisticsFilter(ValueFilter):
+    schema = type_schema("runtime-statistics", rinherit=ValueFilter.schema)
+    permissions = ("athena:GetQueryRuntimeStatistics",)
+
+    def process(self, resources, event=None):
+        client = local_session(self.manager.session_factory).client("athena")
+        for r in resources:
+            if "QueryExecutionId" not in r:
+                # Ensure we have full detail loaded via batch detail
+                continue
+            # Fetch on-demand and attach for filtering
+            try:
+                resp = client.get_query_runtime_statistics(
+                    QueryExecutionId=r["QueryExecutionId"]
+                )
+                stats = resp.get("QueryRuntimeStatistics")
+                if stats is not None:
+                    r["QueryRuntimeStatistics"] = stats
+            except ClientError:
+                # If API is not available/authorized, leave as-is and let matching fail
```

Comment:
so depends on the error re best behavior, not found or one off error is typical to catch, persistent like not authorized we would typically raise, as its typically a fail on the rest of the policy.